test/functional/createPatcher_matches.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * Copyright 2018 The Incremental DOM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// taze: mocha from //third_party/javascript/typings/mocha
+// taze: chai from //third_party/javascript/typings/chai
+
+import {
+  elementVoid,
+  createPatcher,
+  runPatchInner,
+  patchInner,
+} from '../../index';
+import { text } from '../../src/virtual_elements';
+const {expect} = chai;
+
+describe('createPatcher\'s matches option', () => {
+  let container: HTMLElement;
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(container);
+  });
+
+  describe('default matches', () => {
+    it('should match with the same key and node name', () => {
+      const patch = createPatcher(runPatchInner);
```

Comment:
Seems a bit overkill to allow passing in a patch function when only two will realistically work (runPatchInner/runPatchOuter). Should they just be a part of the config instead?

Reply:
Do you mean something like:

```javascript
enum PatchRunner {
  INNER,
  OUTER,
};
```

```javascript
const patch = createPatcher({
  runner: PatchRunner.INNER,
  matches: () => { ... }
});
```

or perhaps just a boolean `inner`?

My thinking was to future proof this API in case we add more patch types, but I'm not sure that really makes sense.

Reply:
Per our discussion, simplified to expose `createPatchInner` and `createPatchOuter` instead of specifying a runner.